scripts/skills/perks/perk_rf_fresh_and_furious.nut

```diff
@@ -45,11 +48,31 @@ this.perk_rf_fresh_and_furious <- ::inherit("scripts/skills/skill", {
 
 	function isEnabled()
 	{
-		return this.getContainer().getActor().getFatigue() < 0.3 * this.getContainer().getActor().getFatigueMax();
+		return !this.m.RequiresRecover && this.getContainer().getActor().getFatigue() < this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax();
 	}
 
 	function onAfterUpdate( _properties )
 	{
+		// We do it like this instead of checking it in onTurnStart directly because some skills e.g. Encumbrance
+		// may modify the fatigue during onTurnStart, so we want to make sure that we check fatigue after that
+		if (this.m.IsStartingTurn)
+		{
+			this.m.IsStartingTurn = false;
+
+			if (this.getContainer().getActor().getFatigue() == 0)
+			{
+				this.m.RequiresRecover = false;
+			}
+			else
+			{
+				if (this.m.RequiresRecover || this.getContainer().getActor().getFatigue() >= this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax())
```

Comment:
`this.m.RequiresRecover ||` is wrong and also redundant unless you are aiming for performance. 
But setting a bool variable to a new value is not worth improving the performance around.

Reply:
The alternative was the following:
```squirrel
if (this.m.RequiresRecover)
   return;

if (this.getContainer().getActor().getFatigue() >= this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax())
{
     this.m.RequiresRecover = true;
     return;
}
```
I just merged them into a single if because if the bool was already true, setting it again to true doesn't hurt and the performance impact is negligible. But I can change it to this more elaborate code setup if you prefer this.

Reply:
I would just remove that condition outright. Currently it is redundant and I don't see that changing any time soon.
So just
```
if (this.getContainer().getActor().getFatigue() >= this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax())
{
     this.m.RequiresRecover = true;
     return;
}
```

Reply:
But we have to return from the function if RequiresRecover is already true. We can't remove that condition.

Reply:
I don't see the issue in not returning early. The second half of this elabtore `onAfterUpdate` body is gated by `this.isEnabled()` which checks `RequiresRecover`